ng/js/controllers/new_posting.js

```diff
@@ -0,0 +1,270 @@
+require('angular').module('ays')
+  .controller('CreatePostingController', function ($scope, $sails, $timeout, $state) {
+    $scope.categoryOptions = [];
+    $scope.categoryDescription = '';
+    $scope.titleErrorMessages = [];
+    $scope.priceErrorMessages = [];
+    $scope.descriptionErrorMessages = [];
+    $scope.categoryErrorMessages = [];
+
+    $scope.allInputValidated = false;
+    $scope.titleValidated = false;
+    $scope.pricesValidated = false;
+    $scope.descriptionValidated = false;
+    $scope.categoryValidated = false;
+    $scope.isSubmitting = false;
+
+    // Populate category options based on what server has
+    $sails.get('/api/category')
+      .then(function (response) {
+        var i = 0;
+        for (i = 0; i < response.data.length; i++) {
+          $scope.categoryOptions.push(response.data[i].description);
+        }
+      });
+
+    $scope.newPosting = {
+      sellerID: '',
+      id: '',
+      title: '',
+      categoryID: '',
+      image: null,
+      description: '',
+      price: {
+        minimum: '',
+        maximum: '',
+      },
+      date: '',
+      tags: '',
+      skills: '',
+      active: false,
+    };
+
+    // Simple helper function that resets the newposting object to default values.
+    function getDisplayObject() {
+      return {
+        sellerID: '',
+        id: '',
+        title: '',
+        categoryID: '',
+        image: null,
+        description: '',
+        price: {
+          minimum: '',
+          maximum: '',
+        },
+        date: '',
+        tags: '',
+        skills: '',
+        active: false,
+      };
+    }
+
+    // Submit the form. Only gets called after validation occurs.
+    $scope.submitForm = function () {
+      $scope.newPosting.categoryID = $scope.categoryOptions.indexOf($scope.categoryDescription);
```

Comment:
The ID of the category has no inherent correlation to its position in the array returned. The server caches them as a map. The options being returned in ID order with no missing indexes could easily change in the future. Like we might decide they should be shown in alphabetical order or that the Rides category can only be shown on some campuses (both extremely likely).